File: test/stack/max-stack.test.ts
import { MaxStack } from '../../src/heap/max-stack';

describe('max stack', () => {
  test('max stack - test case 1', async () => {
    const stack = new MaxStack();

    stack.push(5);
    stack.push(1);
    stack.push(5);

    expect(stack.top()).toBe(5);
    expect(stack.popMax()).toBe(5);
    expect(stack.top()).toBe(1);
    expect(stack.peekMax()).toBe(5);
    expect(stack.pop()).toBe(1);
    expect(stack.top()).toBe(5);
  });
});

File: test/leetcode/heap/max_stack_test.py
import leetcode.heap.max_stack as lc


stack = lc.MaxStack()


def test_case_1():
    stack.push(5)
    stack.push(1)
    stack.push(5)

    assert stack.top() == 5
    assert stack.popMax() == 5
    assert stack.top() == 1
    assert stack.peekMax() == 5
    assert stack.pop() == 1
    assert stack.top() == 5
